tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/TeamMailboxMemberGetMethod.scala

```diff
@@ -0,0 +1,77 @@
+
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, LINAGORA_TEAM_MAILBOXES)
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, TeamMailboxMemberGetRequest] =
+    Serializer.deserializeGetRequest(invocation.arguments.value)
+      .asEither.left.map(ResponseSerializer.asException)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier],
+                         invocation: InvocationWithContext,
+                         mailboxSession: MailboxSession,
+                         request: TeamMailboxMemberGetRequest): Publisher[InvocationWithContext] =
+    getTeamMailboxMemberResponse(mailboxSession.getUser, request)
+      .map(response => Invocation(
+        methodName = methodName,
+        arguments = Arguments(Serializer.serializeGetResponse(response).as[JsObject]),
+        methodCallId = invocation.invocation.methodCallId))
+      .map(InvocationWithContext(_, invocation.processingContext))
+
+  private def getTeamMailboxMemberResponse(username: Username, request: TeamMailboxMemberGetRequest): SMono[TeamMailboxMemberGetResponse] = {
+    request.ids match {
+      case None => getMembersOfAllTeamMailboxes(username, request.accountId)
+      case Some(ids) => getMembersOfSpecificTeamMailboxes(username, request.accountId, ids)
+    }
+  }
+
+  private def getMembersOfAllTeamMailboxes(username: Username, accountId: AccountId): SMono[TeamMailboxMemberGetResponse] =
+    SFlux.fromPublisher(teamMailboxRepository.listTeamMailboxes(username))
+      .flatMap(teamMailbox =>
+        SFlux(teamMailboxRepository.listMembers(teamMailbox))
+          .collectMap(member => member.username.asString(), member => TeamMailboxMemberRoleDTO(member.role.value.toString))
+          .map(mapMembers => TeamMailboxMemberDTO(teamMailbox.mailboxName.asString(), mapMembers)))
+      .collectSeq()
+      .map(seq => TeamMailboxMemberGetResponse(accountId, seq, Seq.empty))
+
+  private def getMembersOfSpecificTeamMailboxes(username: Username, accountId: AccountId, mailboxNames: Set[String]): SMono[TeamMailboxMemberGetResponse] = {
+    SFlux.fromPublisher(teamMailboxRepository.listTeamMailboxes(username))
+      .collectSeq()
+      .flatMap(teamMailboxes => {
+        val foundTeamMailboxes = teamMailboxes.filter(teamMailbox => mailboxNames.contains(teamMailbox.mailboxName.asString()))
+        val notFoundTeamMailboxNames = mailboxNames.diff(foundTeamMailboxes.map(teamMailbox => teamMailbox.mailboxName.asString()).toSet).toSeq
+        SFlux.fromIterable(foundTeamMailboxes)
+          .flatMap(teamMailbox =>
+            SFlux(teamMailboxRepository.listMembers(teamMailbox))
+              .collectMap(member => member.username.asString(), member => TeamMailboxMemberRoleDTO(member.role.value.toString))
+              .map(mapMembers => TeamMailboxMemberDTO(teamMailbox.mailboxName.asString(), mapMembers)))
+          .collectSeq()
```

Comment:
From the `flatMap(teamMailbox`  to the collectSeq, duplicated with def above. Extract it into a private def